src/cf.data.pre

```diff
@@ -7112,11 +7112,28 @@ LOC: Config.visibleHostname
 DEFAULT: none
 DEFAULT_DOC: Automatically detect the system host name
 DOC_START
-	If you want to present a special hostname in error messages, etc,
-	define this.  Otherwise, the return value of gethostname()
-	will be used. If you have multiple caches in a cluster and
-	get errors about IP-forwarding you must set them to have individual
-	names with this setting.
+	The hostname of this proxy.
+
+	Hosts connected to the Internet or handling messages which travel
+	over the Internet are required by RFC 1123 to have a name which
+	can be resolved in DNS (an FQDN).
+
+	If not set, auto-detection of the system host name will be performed
+	based on the following details (in this order):
+		- rDNS name for the first listening port address
+		- the name returned by gethostname(2)
+		- the name "localhost"
+
+	This name will be used in generated responses sent to clients
+	for objects they need to display the proxy name. For example;
+	error pages.
+
+	This name will also be used in generated URLs sent to clients for
+	objects they need to download directly from Squid. For example;
+	CSS stylesheets, FTP icons.
+
+	See unique_hostname directive if you have multiple proxies in a
+	cluster which need to share a visible hostname.
```

Comment:
The subject here is a plural 'proxies'. Plural of "that" is "those" - so: "proxies in a cluster those need to share a". Does not make sense.

Reply:
> Plural of "that" is "those"

... when used as a [demonstrative](https://en.wikipedia.org/wiki/Demonstrative). In our context, `that` is not a demonstrative. It is a pronoun that introduces a relative [clause](https://en.wikipedia.org/wiki/English_relative_clauses). Unlike demonstratives, in our context, `that` is a part of speech that does not have singular and plural forms at all -- that is why you intuitively know that `proxies those need to share` simply "does not make sense" here.

The reason we should use `that` instead of `which` in our context is because the word in question introduces a [_restrictive_](https://en.wikipedia.org/wiki/English_relative_clauses#Restrictive_or_non-restrictive_relative_clauses) clause rather than a non-restrictive one. Restrictive clauses [should](https://www.grammarly.com/blog/which-vs-that/) be introduced by `that`.